src/PHPUnit/BootFoundryOnDataProviderMethodCalled.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Zenstruck\Foundry\PHPUnit;
+
+use App\Infrastructure\Common\Symfony\Kernel;
+use PHPUnit\Event;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Zenstruck\Foundry\Configuration;
+use Zenstruck\Foundry\Test\UnitTestConfig;
+
+/**
+ * @author Nicolas PHILIPPE <nikophil@gmail.com>
+ */
+final class BootFoundryOnDataProviderMethodCalled implements Event\Test\DataProviderMethodCalledSubscriber
+{
+    public function __construct(
+        private KernelInterface $kernel
+    )
+    {
+    }
+
+    public function notify(Event\Test\DataProviderMethodCalled $event): void
+    {
+        if (is_a($event->testMethod()->className(), KernelTestCase::class, allow_string: true)) {
+            static $kernelIsBooted = false;
+
+            if (!$kernelIsBooted) {
+                $this->kernel->boot();
+                $kernelIsBooted = true;
+            }
```

Comment:
thanks to this, we can use service factories in data providers.

One downside of this, is that we're kinda always boot the kernel, even for tests like `vendor/bin/phpunit --filter SomeUnitTest`, because all data providers are always called. This may result in a longer bootstrap phase.

I'm wondering if we should not enable this in Foundry's config, WDYT?